### DISCUSSION TITLE
immortalwrt固件homeproxy睡一觉醒来就无法访问国外了

### DISCUSSION THREAD

**Discussion by soapmancn** [immortalwrt/homeproxy#346] [General]
### Bug description

immortalwrt固件homeproxy睡一觉醒来第二天国外网站就无法访问了，查看状态还是运行中，检测服务状态国内是通过，国外是未通过，重新关闭homeproxy再开启，国外才会检测通过并访问。

### HomeProxy configuration

config homeproxy 'infra'
        option __warning 'DO NOT EDIT THIS SECTION, OR YOU ARE ON YOUR OWN!'
        option common_port '22,53,80,143,443,465,853,873,993,995,8080,8443,9418'
        option mixed_port '5330'
        option redirect_port '5331'
        option tproxy_port '5332'
        option dns_port '5333'
        option china_dns_port '5334'
        option tun_name 'singtun0'
        option tun_addr4 '172.19.0.1/30'
        option tun_addr6 'fdfe:dcba:9871::1/126'
        option tun_mtu '9000'
        option tun_gso '0'
        option table_mark '100'
        option self_mark '100'
        option tproxy_mark '101'
        option tun_mark '102'

config homeproxy 'config'
        option main_node 'd23d54789413b0aa461d4ce962b730b6'
        option main_udp_node 'same'
        option dns_server '8.8.8.8'
        option routing_mode 'bypass_mainland_china'
        option routing_port 'common'
        option proxy_mode 'tun'
        option ipv6_support '1'
        list china_dns_server '223.5.5.5'
        list china_dns_server '119.29.29.29'

config homeproxy 'control'
        option lan_proxy_mode 'disabled'
        list wan_proxy_ipv4_ips '91.105.192.0/23'
        list wan_proxy_ipv4_ips '91.108.4.0/22'
        list wan_proxy_ipv4_ips '91.108.8.0/22'
        list wan_proxy_ipv4_ips '91.108.16.0/22'
        list wan_proxy_ipv4_ips '91.108.12.0/22'
        list wan_proxy_ipv4_ips '91.108.20.0/22'
        list wan_proxy_ipv4_ips '91.108.56.0/22'
        list wan_proxy_ipv4_ips '149.154.160.0/20'
        list wan_proxy_ipv4_ips '185.76.151.0/24'
        list wan_proxy_ipv6_ips '2001:67c:4e8::/48'
        list wan_proxy_ipv6_ips '2001:b28:f23c::/48'
        list wan_proxy_ipv6_ips '2001:b28:f23d::/48'
        list wan_proxy_ipv6_ips '2001:b28:f23f::/48'
        list wan_proxy_ipv6_ips '2a0a:f280::/32'

config homeproxy 'routing'
        option sniff_override '1'
        option default_outbound 'direct-out'
        option tun_gso '0'
        option tcpip_stack 'system'
        option udp_timeout '300'
        option bypass_cn_traffic '0'

config homeproxy 'dns'
        option dns_strategy 'prefer_ipv4'
        option default_server 'default-dns'

config homeproxy 'subscription'
        option auto_update '0'
        option allow_insecure '0'
        option packet_encoding 'xudp'
        option update_via_proxy '0'
        option filter_nodes 'blacklist'
        list filter_keywords '重置|到期|过期|剩余|套餐'
        list filter_keywords 'Expiration|Remaining'

config homeproxy 'server'
        option enabled '0'
        option auto_firewall '0'

config dns_rule 'nodes_domain'
        option label 'NodesDomain'
        option enabled '1'
        option mode 'default'
        list outbound 'any-out'
        option server 'default-dns'

config node 'd23d54789433b0aa461d4ce962b739b7'
        option label 'Misaka-Hysteria2'
        option type 'hysteria2'
        option address '149.135.67.55'
        option port '6666'
        option password '66669d20'
        option tls '1'
        option tls_sni 'www.bing.com'
        option tls_insecure '1'

### sing-box configuration

> root@openwrt:/var# cat /var/run/homeproxy/sing-box-c.json
{
        "log": {
                "disabled": false,
                "level": "warn",
                "output": "/var/run/homeproxy/sing-box-c.log",
                "timestamp": true
        },
        "dns": {
                "servers": [
                        {
                                "tag": "default-dns",
                                "address": "202.96.128.166",
                                "detour": "direct-out"
                        },
                        {
                                "tag": "system-dns",
                                "address": "local",
                                "detour": "direct-out"
                        },
                        {
                                "tag": "block-dns",
                                "address": "rcode://name_error"
                        },
                        {
                                "tag": "main-dns",
                                "address": "tcp://8.8.8.8",
                                "detour": "main-out"
                        }
                ],
                "disable_cache": false,
                "disable_expire": false,
                "independent_cache": false,
                "final": "main-dns"
        },
        "inbounds": [
                {
                        "type": "direct",
                        "tag": "dns-in",
                        "listen": "::",
                        "listen_port": 5333
                },
                {
                        "type": "mixed",
                        "tag": "mixed-in",
                        "listen": "::",
                        "listen_port": 5330,
                        "sniff": true,
                        "sniff_override_destination": true,
                        "set_system_proxy": false
                },
                {
                        "type": "tun",
                        "tag": "tun-in",
                        "interface_name": "singtun0",
                        "address": [
                                "172.19.0.1/30",
                                "fdfe:dcba:9876::1/126"
                        ],
                        "mtu": 9000,
                        "gso": false,
                        "auto_route": false,
                        "stack": "system",
                        "sniff": true,
                        "sniff_override_destination": true
                }
        ],
        "outbounds": [
                {
                        "type": "direct",
                        "tag": "direct-out"
                },
                {
                        "type": "block",
                        "tag": "block-out"
                },
                {
                        "type": "dns",
                        "tag": "dns-out"
                },
                {
                        "type": "hysteria2",
                        "tag": "main-out",
                        "server": "141.135.67.51",
                        "server_port": 6666,
                        "password": "66699d20",
                        "tls": {
                                "enabled": true,
                                "server_name": "www.bing.com",
                                "insecure": true
                        }
                }
        ],
        "route": {
                "rules": [
                        {
                                "inbound": "dns-in",
                                "outbound": "dns-out"
                        }
                ],
                "rule_set": [
                ],
                "auto_detect_interface": true,
                "final": "main-out"
        }
}

### sing-box log

no log

### OpenWrt/ImmortalWrt release

23.05.4 r28061-399f9a1db3

### Device

QEMU QEMU Virtual Machine

### Terms

- [X] I confirm that the HomeProxy I installed is from official source, like GitHub artifacts or official ImmortalWrt opkg feeds.
- [X] I confirm that the Homeproxy I installed does not contain Clash mode support.
- [X] I confirm that the version of OpenWrt/ImmortalWrt I installed is >= 23.05.
- [X] I confirm that I have installed the latest version of HomeProxy and sing-box.
- [X] I confirm that I have read the sing-box documentation, understand the meaning of all the configuration items I added.
- [X] I confirm that I have not mixed iptables and nftables rules.
- [X] I confirm that I have not modified system DNS settings, or enabled any other DNS servers like MosDNS and SmartDNS.
- [ ] I confirm that I have selected all terms **blindly**.

**Comment by soapmancn** [comment]
换成24.10固件也是这样，只装了homeproxy跟ddns-go插件，一段时间不访问外网，homeproxy就变成未运行了。

**Comment by KevinMX** [comment]
刚刚蹲到自动更新订阅和 geodata，复现了，具体表现为：
开启自动更新订阅和地理数据后，更新好订阅数据之后，必须手动重新启动 homeproxy 服务。
用的 URLSelect，里面只有一个 VLESS 节点。

```log
2025-02-24 16:16:51 [DAEMON] sing-box 1.11.3 started.
2025-02-24 16:17:08 [CHINA_IP4] Local version: 20250127031156, latest version: 20250224031805.
2025-02-24 16:17:11 [CHINA_IP4] Successfully updated.
2025-02-24 16:17:12 [CHINA_IP6] Local version: 20250127031156, latest version: 20250224031805.
2025-02-24 16:17:13 [CHINA_IP6] Successfully updated.
2025-02-24 16:17:15 [CHINA_LIST] Local version: 202501262210, latest version: 202502232211.
2025-02-24 16:17:16 [CHINA_LIST] Successfully updated.
2025-02-24 16:17:17 [GFW_LIST] Local version: 202501262210, latest version: 202502232211.
2025-02-24 16:17:17 [GFW_LIST] Successfully updated.
#（这里自动触发了定时更新订阅和 geodata）
2025-02-25 02:00:00 [CHINA_IP4] Current version: 20250224031805.
2025-02-25 02:00:00 [CHINA_IP4] You're already at the latest version.
2025-02-25 02:00:01 [CHINA_IP6] Current version: 20250224031805.
2025-02-25 02:00:01 [CHINA_IP6] You're already at the latest version.
2025-02-25 02:00:02 [GFW_LIST] Current version: 202502232211.
2025-02-25 02:00:02 [GFW_LIST] You're already at the latest version.
2025-02-25 02:00:02 [CHINA_LIST] Current version: 202502232211.
2025-02-25 02:00:02 [CHINA_LIST] You're already at the latest version.
2025-02-25 02:00:02 [SUBSCRIBE] Stopping service...
2025-02-25 02:00:08 [DAEMON] Service stopped.
#（这里往下手动启动了 homeproxy 服务）
[33mWARN[0m[0000] legacy special outbounds is deprecated in sing-box 1.11.0 and will be removed in sing-box 1.13.0, checkout documentation for migration: https://sing-box.sagernet.org/migration/#migrate-legacy-special-outbounds-to-rule-actions
2025-02-25 02:06:17 [DAEMON] sing-box 1.11.3 started.
2025-02-25 02:06:35 [SUBSCRIBE] Failed to fetch resources from https://my-sub-url.
2025-02-25 02:06:35 [SUBSCRIBE] Failed to update subscriptions: no valid node found.
2025-02-25 02:06:35 [SUBSCRIBE] Starting service...
[33mWARN[0m[0000] legacy special outbounds is deprecated in sing-box 1.11.0 and will be removed in sing-box 1.13.0, checkout documentation for migration: https://sing-box.sagernet.org/migration/#migrate-legacy-special-outbounds-to-rule-actions
2025-02-25 02:06:39 [DAEMON] sing-box 1.11.3 started.
```

应该与其他设置无关，暂时就没贴配置。
应该是个 Regression，以前版本印象中没有这个问题。
虽然我确实很久没用了，有需要补充信息的话随时喊我（

**Comment by KevinMX** [comment]
走 Firmware Selector 重新生成固件不保留配置刷入后暂时未能复现
变动是去掉了节点订阅，仅保留了 Geodata
继续观测中

**Comment by initzhu** [comment]
节点设置-订阅-勾选自动更新
建议选中使用代理更新

如果您更新固件比较频繁 请在“备份与升级-配置”中加入
/etc/homeproxy/

**Comment by 13311031900** [comment]
我固件也是这个版本，升级最新的homeproxy版本后更新资源管理里的各个规则都无法更新，我的是nanopi r4se，新固件版本24.10这个机器用不了，刷了就进不了系统。

**Comment by libriunk** [comment]
> 我固件也是这个版本，升级最新的homeproxy版本后更新资源管理里的各个规则都无法更新，我的是nanopi r4se，新固件版本24.10这个机器用不了，刷了就进不了系统。

这是两个问题
 你r4se需要修改固件 使用selector 将8*68改成8*69
hp用不了可以试试换个dns解析

**Comment by guziwen-lab** [comment]
相同的问题，请问解决了吗？
